cmd/csi-provisioner/csi-provisioner.go

```diff
@@ -460,10 +461,17 @@ func main() {
 		}
 
 		// We only need objects from our own namespace. The normal factory would give
-		// us an informer for the entire cluster.
+		// us an informer for the entire cluster. We can further restrict the
+		// watch to just those objects with the right labels.
 		factoryForNamespace = informers.NewSharedInformerFactoryWithOptions(clientset,
 			ctrl.ResyncPeriodOfCsiNodeInformer,
 			informers.WithNamespace(namespace),
+			informers.WithTweakListOptions(func(lo *metav1.ListOptions) {
```

Comment:
I'm wondering how easy it is to detect if the informer doesn't fetch objects we expect, and whether we need to unit test this.

If informer logic is wrong, two things could happen:

* Missing items that are supposed to be included
  * May try to create an existing CSIStorageCapacity object. In the logic today this is fine, because nothing is done after object creation (besides logging / error reporting), and the creation error would give us a hint.
* Includes more things than it should
  * No failures, just more network traffic.

The latter case is hard to detect. So maybe a unit test is worthwhile (if it's not too much effort)?

Reply:
Actually since this is just an options configuration, the logic is simple enough, so it's probably low risk. Totally up to you if you want to unit test this.

Reply:
I'd prefer to not test this. I suspect that it wouldn't be easy to test comprehensively and in the end is more about correct implementation of the informer and apiserver filtering than it is about correct configuration.